### PR TITLE
Add Kubernetes Auth Method

### PR DESCRIPTION
**Pull request by jsporna**
Add Kubernetes Auth Method based on Vault API documentation with unit and integration tests.
https://www.vaultproject.io/api/auth/kubernetes/index.html

Add function to dummy check PEM-formatted certificate / public key

Fix for comparision `recovery_threshold` and `recovery_shares` during initialization.

### CODE REVIEW

**Comment by jeffwecan** [comment]
Howdy @jsporna, thanks for the PR! I'll try to give this a good once over over the next day or so and get back to you. 👍

**Comment by jsporna** [comment]
Hello @jeffwecan 
Is there any chance that the auth method for kubernetes will come to the next release?
I have in plans to implement also auth method for AWS :)

**Comment by jeffwecan** [comment]
Sure! We're overdue for a release so I'll aim to get a new release published by the end of the week (after I get a chance to review this PR and some of the other outstanding PRs here ☺).

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,15 @@
 # Changelog
 
+## 0.8.0 (March 26th, 2019)
```
Comment (jeffwecan) [inline CHANGELOG.md]: Can you change the date here to "UNRELEASED" so we know to go back and update it when the actual release is cut please?

Comment (jeffwecan) [reply CHANGELOG.md]: Actually I grab this when updating the change log before the release is finalized, so we're good here. 👍

**Pull request merged**